server/src/test/scala/actors/minutes/FlightsRouterActorSpec.scala

```diff
@@ -259,4 +265,47 @@ class FlightsRouterActorSpec extends CrunchTestLike {
       }
     }
   }
+
+  "Concerning persistence of flights" >> {
+    "Given a router, I should see updates sent to it are persisted" >> {
+      val lookups = FlightLookups(system, myNow, queuesByTerminal = Map(T1 -> Seq(EeaDesk, NonEeaDesk, EGate)), updatesSubscriber = TestProbe("").ref)
+      val router = lookups.flightsActor
+
+      val scheduled = "2021-06-01T00:00"
+      val arrival = ArrivalGenerator.arrival(iata = "BA0001", schDt = scheduled, terminal = T1)
+      val requestForFlights = GetFlights(SDate(scheduled).millisSinceEpoch, SDate(scheduled).addHours(6).millisSinceEpoch)
+
+      "When I send it a flight with no splits" >> {
+        val eventualFlights = router
+          .ask(ArrivalsDiff(Iterable(arrival), Iterable()))
+          .flatMap(_ => runAndCombine(
+            router
+              .ask(requestForFlights)
+              .mapTo[Source[FlightsWithSplits, NotUsed]])
+            .map(_.flights.values.headOption))
+
+        val result = Await.result(eventualFlights, 1 second)
+
+        result === Option(ApiFlightWithSplits(arrival, Set(), lastUpdated = Option(myNow().millisSinceEpoch)))
+      }
+
+      "When I send it a flight with no splits, followed by its splits" >> {
+        val splits = Splits(Set(ApiPaxTypeAndQueueCount(EeaNonMachineReadable, EeaDesk, 1, None, None)), Historical, None, PaxNumbers)
+        val eventualFlights = router
+          .ask(ArrivalsDiff(Iterable(arrival), Iterable()))
+          .flatMap(_ => router
+            .ask(SplitsForArrivals(Map(arrival.unique -> Set(splits))))
+            .flatMap(_ => runAndCombine(
+              router
+                .ask(requestForFlights)
+                .mapTo[Source[FlightsWithSplits, NotUsed]])
+              .map(_.flights.values.headOption)))
+
+
+        val result = Await.result(eventualFlights, 1 second)
+
+        result === Option(ApiFlightWithSplits(arrival, Set(splits), lastUpdated = Option(myNow().millisSinceEpoch)))
+      }
+    }
```

Comment:
Do we need a test for the inverse order - splits then flight?

Reply:
This is a good question, and it actually won't work if we send the split first because it won't have a flight to save it against.

However, this situation won't arise any longer since every time we crunch we first look up manifests for all flights and then convert them to splits before saving them against the flight.